x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/SubmitAsyncSearchRequest.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.search.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * A request to track asynchronously the progress of a search against one or more indices.
+ *
+ * @see AsyncSearchResponse
+ */
+public class SubmitAsyncSearchRequest extends ActionRequest {
+    public static long MIN_KEEP_ALIVE = TimeValue.timeValueHours(1).millis();
+
+    private TimeValue waitForCompletion = TimeValue.timeValueSeconds(1);
+    private boolean cleanOnCompletion = true;
+    private TimeValue keepAlive = TimeValue.timeValueDays(5);
+
+    private final SearchRequest request;
+
+    /**
+     * Creates a new request
+     */
+    public SubmitAsyncSearchRequest(String... indices) {
+        this(new SearchSourceBuilder(), indices);
+    }
+
+    /**
+     * Creates a new request
+     */
+    public SubmitAsyncSearchRequest(SearchSourceBuilder source, String... indices) {
+        this.request = new SearchRequest(indices, source);
+        request.setCcsMinimizeRoundtrips(false);
+        request.setPreFilterShardSize(1);
+        request.setBatchedReduceSize(5);
+        request.requestCache(true);
```

Comment:
I think it's ok to leave them open, some of them are already forbidden (CcsMinimizeRoundtrips and scroll) since they break the API but I don't see why we should prevent the other ones since this API should expose the same option than a normal search.

Reply:
agreed, my comment focused on stuff that if changed, may affect how the API works e.g. pre_filter_shard_size, batched_reduced_size. Can you show me where we reject ccs minimize roundtrips set to false?